{Release} Upgrade to Azure CLI 2.57.0 (#28281)
Azure/azure-cli: src/azure-cli/HISTORY.rst

Release History
===============

2.57.0
++++++

**AKS**

* `az aks upgrade`: Add forceupgrade settings to aks stable cli (#28200)
* `az aks mesh`: Add `az aks mesh` commands to manage Azure Service Mesh in given cluster (#28206)
* `az aks nodepool add/update/upgrade`: Add new parameter `--node-soak-duration` to wait after draining a node and before reimaging it and moving on to next node (#27604)

**App Service**

* `az functionapp create`: Use app insights connection string instead of instrumentation key (#27803)
* `az webapp create`: Add new parameter `--acr-use-identity` to configure pull image from ACR using MSI when creating a container web app (#28209)
* `az webapp up`: Add `--track-status` arg to use deploymentstatus API (#28235)
* `az webapp deployment source config-zip`: Add `--track-status` arg to use deploymentstatus API (#28235)
* `az functionapp create`: Only allow apps with functions version 4 (#28128)

**Backup**

> * Add support for disk restore in edge-zone backups (#28150)
> * Support for Taiwan North and Taiwan North-West Cross region restore (#28150)

Reviewer: ```suggestion
* `az backup restore restore-disks`: Add support for disk restore in edge-zone backups (#28150)
* `az backup restore restore-disks`: Support for Taiwan North and Taiwan North-West Cross region restore (#28150)
```

**Billing**

* `az billing account`: Add new command `list-invoice-section` to support listing invoice section (#28214)
* `az billing account`: Add new command group `invoice-section` to support managing account invoice section (#28214)
* `az billing`: Add new command `enrollment-account` to support managing enrollment account (#28214)
* `az billing invoice section`: Add new command `initiate-transfer` to support intiating transfer (#28214)
* `az billing profile`: Add new command `reservation list ` to support managing reservation (#28214)
* `az billing`: Add new command group `transfer` to support managing transfer (#28214)

**Compute**

* `az image builder error-handler`: Add group to manage error handling options upon a build failure (#28106)
* `az sshkey create`: Add parameter `--encryption-type` to specify the encryption type of SSH keys to be generated (#28143)
* `az vm monitor log show`: Deprecate `azure-loganalytics` and apply `azure-monitor-query` (#28199)

> * `az vm create` and `az vmss create`: Support Trusted Launch as default deployment option (#28222)

Reviewer: ```suggestion
* `az vm/vmss create`: Support Trusted Launch as default deployment option (#28222)
```

**Consumption**

* Fix #20995 #23825: Update `az consumption` API version (#27833)

**Containerapp**

* `az containerapp update`: Fix issue for minReplicas in `--yaml` or `--min-replicas` is not set when the value is 0 (#28163)
* `az containerapp up`: Fix issue when registry creds are provided for ACRs using `containerapp up` command (#28217)

**Key Vault**

* `az keyvault create`: Add warning for upcoming breaking change that `--enable-rbac-authorization` will default to `true` (#28178)
* `az keyvault secret set-attributes`: Fix 'Datetime with no tzinfo will be considered UTC' warning (#28138)

**Monitor**

* `az monitor metrics`: Add `list-sub` and `list-sub-definition` (#28201)

> 

Reviewer 2: ```suggestion
`az monitor metrics list`: Fix `--top` help message (#28114)
```

> 

Reviewer 2: ```suggestion
* `az monitor metrics list`: Fix `--top` help message (#28114)
```

> * `az monitor metrics list`: Fix `--top` help message (#28114)

Reviewer 3: ```suggestion
* `az monitor metrics list`: Fix `--top` help message (#28114)

```

* `az network express-route port authorization`: Support to manage authorization with express route port (#28149)
* `az network private-link-service list-auto-approved`: Support to list all auto-approved private link services (#28149)
* `az network public-ip ddos-protection-statu show`: Support to get the DDoS protection status of public IP address (#28149)
* `az network vnet-gateway`: Support ExpressRoute SKU (#28219)
* `az network lb create`: Support cross-subscription resource ID (#28247)

**Packaging**

* Support Windows ZIP package (#27911)

**RDBMS**

* `az postgres flexible-server private-endpoint-connection`: Add support for private endpoint connection commands like list, show, approve, reject, delete for PostgreSQL flexible server (#28142)
* `az postgres flexible-server private-link-resource`: Add support for private link resource commands like list, show for PostgreSQL flexible server (#28142)
* `az postgres flexible-server replica stop-replication`: Stop replication to a read replica and make it a read/write server. This command is deprecated. Use `az postgres flexible-server replica promote` instead. (#28189)

**Redis**

> * Add support for flush operation (#27599)
> * Add support for Microsoft Entra Authentication (#27599)
> * Add support for update channels (#27599)

Reviewer: ```suggestion
* `az redis flush`: Add support for flush operation (#27599)
* `az redis create`: Add support for Microsoft Entra Authentication (#27599)
* `az redis access-policy/access-policy-assignment`: New command groups to manage access policy (#27599)
```
